test/case/ietf_system/upgrade/test.py

```diff
@@ -17,11 +20,36 @@
     "bundles"
 )
 
+bootloader=None
 PKGPATH = os.path.join(
     BUNDLEDIR,
     "package"
 )
 
+class Uboot:
+    def __init__(self, ssh):
+        self.ssh=ssh
+
+    def get_boot_order(self):
+        order=self.ssh.runsh("sudo fw_printenv BOOT_ORDER").stdout.split("=")
+        return order[1]
+
+    def set_boot_order(self, order):
+        return self.ssh.run(f"sudo fw_setenv BOOT_ORDER '{order}'".split()).returncode
+
+class Grub:
+    def __init__(self, ssh):
+        self.ssh = ssh
+
+    def get_boot_order(self):
+        lines=self.ssh.runsh("grub-editenv /mnt/aux/grub/grubenv list").stdout.split("\n")
+        for line in lines:
+            if "ORDER" in line:
+                return line.split("=")[1]
+
+    def set_boot_order(self, order):
+        return self.ssh.run(f"sudo grub-editenv /mnt/aux/grub/grubenv set ORDER='{order}'".split()).returncode
+
 with infamy.Test() as test:
```

Comment:
We might want to add some way for a test implementation to register cleanup functions. Such that the boot order is restored even if we break the test in the middle (e.g., pressing `C-c`)